### PR TITLE
feat: add bootstrap token package

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

File: internal/pkg/token/token.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package token
+
+import (
+	"github.com/google/uuid"
+)
+
+const (
+	BootstrapTTL uuid.Time = 3600
```
Comment (smira) [inline internal/pkg/token/token.go]: should this be controlled via userdata as setting?

File: pkg/userdata/token/token.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (smira) [inline pkg/userdata/token/token.go]: looks like duplicate code?

Comment (andrewrynhard) [reply pkg/userdata/token/token.go]: Fixed.

File: internal/pkg/token/token.go (hunk 1 above)
Comment (andrewrynhard) [reply internal/pkg/token/token.go]: 🤔 ... @bradbeam what do you think?

Comment (andrewrynhard) [reply internal/pkg/token/token.go]: I'm ok either way. I think we wanted to not give the user too much power over this just yet.

Comment (bradbeam) [reply internal/pkg/token/token.go]: we talked about it a bit. we'll start with hardcoded for now and adjust as needed. 1h seems generous enough to do an initial bootstrap of the cluster and that's really the only intent of the bootstrap token.

**Pull request merged**